### PR TITLE
build: generate project code

### PR DESCRIPTION
**Pull request by dvesperini**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by dvesperini** [comment]
@handadi @hkantare 
Please review

File: ibm/service/project/resource_ibm_project.go [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+		return diag.FromErr(err)
+	}
+
+	updateProjectOptions := &projectv1.UpdateProjectOptions{}
+
+	updateProjectOptions.SetID(d.Id())
+
+	hasChange := false
+
+	if hasChange {
```
Comment (kavya498) [inline ibm/service/project/resource_ibm_project.go:366]: It is never going to enter this loop.
kindly check for update on arguments before this if condition

Ref: https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/service/vpc/resource_ibm_is_vpc.go#L831

File: ibm/service/project/resource_ibm_project.go [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+func resourceIbmProjectDelete(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	projectClient, err := meta.(conns.ClientSession).ProjectV1()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	deleteProjectOptions := &projectv1.DeleteProjectOptions{}
+
+	deleteProjectOptions.SetID(d.Id())
+
```
Comment (kavya498) [inline ibm/service/project/resource_ibm_project.go:386]: We would encourage you to provide `destroy` flag feature like in api.
Is there any specific reason for not including it?

File: website/docs/r/project.html.markdown [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+# ibm_project
+
+Provides a resource for project. This allows project to be created, updated and deleted.
+
+## Example Usage
+
+```hcl
+resource "ibm_project" "project_instance" {
+  configs {
+		id = "id"
```
Comment (kavya498) [inline website/docs/r/project.html.markdown:18]: Correct me if I'm wrong here. 
id here is project id? If so how do we provide project id during creation of project.

File: website/docs/r/project.html.markdown [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+		name = "name"
+		labels = [ "labels" ]
+		description = "description"
+		locator_id = "locator_id"
```
Comment (kavya498) [inline website/docs/r/project.html.markdown:22]: kindly provide working example
kkindly provide valid values to the argument to make this doc more user friendly.

File: website/docs/r/project.html.markdown [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+		* `event` - (String) The event name.
+		* `event_id` - (String) The unique ID of a project.
+		  * Constraints: The maximum length is `128` characters. The value must match regular expression `/^[\\.\\-0-9a-zA-Z]+$/`.
+	* `cumulative_needs_attention_view_err` - (String) True to indicate the fetch of needs attention items that failed.
+	* `event_notifications_crn` - (String) The CRN of the event notifications instance if one is connected to this project.
+	* `location` - (String) The location of where the project was created.
+	* `resource_group` - (String) The resource group of where the project was created.
+	* `state` - (String) The project status value.
+	  * Constraints: The maximum length is `64` characters. The minimum length is `1` character. The value must match regular expression `/^(CREATING|CREATING_FAILED|UPDATING|UPDATING_FAILED|READY)$/`.
+
```
Comment (kavya498) [inline website/docs/r/project.html.markdown:101]: You can remove 
`# Provider Configuration`
`### Static credentials`
`### Environment variables` sections from all project docs as it is generic doc and is available in provider page - https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs#authentication

File: ibm/service/project/data_source_ibm_event_notification.go [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+	if err = d.Set("updated_at", flex.DateTimeToString(getEventNotificationsIntegrationResponse.UpdatedAt)); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting updated_at: %s", err))
+	}
+
+	if err = d.Set("topic_count", flex.IntValue(getEventNotificationsIntegrationResponse.TopicCount)); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting topic_count: %s", err))
+	}
+
```
Comment (kavya498) [inline ibm/service/project/data_source_ibm_event_notification.go:112]: topic_names is not been set

File: ibm/service/project/data_source_ibm_event_notification_test.go [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIbmEventNotificationDataSourceConfigBasic() string {
+	return fmt.Sprintf(`
+		data "ibm_event_notification" "event_notification_instance" {
+			id = "id"
```
Comment (kavya498) [inline ibm/service/project/data_source_ibm_event_notification_test.go:34]: Kindly provide working example to the testcase.

**Review by kavya498 — CHANGES_REQUESTED** [review]
Kindly, resolve conflicts and attach testcase results of resource and datasources to the PR.

File: go.mod [hunk 8]
```diff
@@ -24,6 +24,7 @@ require (
 	github.com/IBM/keyprotect-go-client v0.9.0
 	github.com/IBM/networking-go-sdk v0.36.0
 	github.com/IBM/platform-services-go-sdk v0.32.1
+	github.com/damianovesperini/platform-services-go-sdk v1.0.9
```
[the remark this answers is not in the document]
Comment (dvesperini) [reply go.mod:27]: cause we don't have our go sdk published under IBM

File: ibm/service/project/data_source_ibm_event_notification.go [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/damianovesperini/platform-services-go-sdk/projectv1"
+)
+
+func DataSourceIbmEventNotification() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIbmEventNotificationRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
```
[the remark this answers is not in the document]
Comment (dvesperini) [reply ibm/service/project/data_source_ibm_event_notification.go:24]: We won't change that id into project_id, we have this naming convention also for CLI and SDK.

File: website/docs/d/project.html.markdown [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+## Argument Reference
+
+Review the argument reference that you can specify for your data source.
+
+* `complete` - (Optional, Boolean) The flag to determine if full metadata should be returned.
+  * Constraints: The default value is `false`.
+* `exclude_configs` - (Optional, Boolean) Only return with the active configuration, no drafts.
+  * Constraints: The default value is `false`.
+* `id` - (Required, Forces new resource, String) The ID of the project, which uniquely identifies it.
```
[the remark this answers is not in the document]
Comment (dvesperini) [reply website/docs/d/project.html.markdown:29]: same as I said in the previous comment.. we want to stay with id

File: ibm/service/project/data_source_ibm_event_notification.go (hunk 9 above)
[the remark this answers is not in the document]
Comment (dvesperini) [reply ibm/service/project/data_source_ibm_event_notification.go:24]: We won't change this.. we have the same approach under SKD,CLI ecc..

File: ibm/conns/config.go [hunk 11]
```diff
@@ -1516,6 +1527,24 @@ func (c *Config) ClientSession() (interface{}, error) {
 		}
 	}
 
+	// Construct an "options" struct for creating the service client.
+	projectClientOptions := &projectv1.ProjectV1Options{
+		Authenticator: authenticator,
+	}
+
```
[the remark this answers is not in the document]
Comment (riccardo-angelilli) [reply ibm/conns/config.go:1535]: done

File: ibm/conns/config.go [hunk 12]
```diff
@@ -1516,6 +1527,24 @@ func (c *Config) ClientSession() (interface{}, error) {
...
+	// Construct the service client.
+	session.projectClient, err = projectv1.NewProjectV1(projectClientOptions)
```
[the remark this answers is not in the document]
Comment (riccardo-angelilli) [reply ibm/conns/config.go:1544]: done

**Comment by dvesperini** [comment]
I'm going to close this because I have opened a new one:
https://github.com/IBM-Cloud/terraform-provider-ibm/pull/4484

**Pull request closed without merging**